doc/specs/tagging.md

```diff
@@ -0,0 +1,147 @@
+# Tagging
+
+## Table of Contents
+
+1. Motivation
+2. Business rules
+3. Use cases
+
+## Motivation
+
+Tagging allows items on the site to be associated with a topic, either broad or granular. Having 
+such a system in place allows users to find exactly what they're looking for with much greater
+ease.
+
+## Business rules
+
+* Tags must come in two options:
+  * Simple tagging, which allows for an unlimited number of tags.
+  * Hierarchal tagging, which allows for sub-tags.
+* The number of tags must have a limit that is configurable on a case-by-case basis. (i.e. a limit
+  of 5 tags for a wiki article, or 2 tags for collages)
+  * In the case of hierarchal tagging, the following options must be configurable:
+    * Required or optional parent tags, minimum number of parent tags, or maximum number of parent tags.
+    * Required or optional child tags, minimum number of child tags, or maximum number of child tags.
+    * Selection of parent tags before child tags required or optional.
+* Each system that uses the tagging packge must have the option to force tags to be chosen from
+  a pre-defined set. (i.e. a forum limiting users to tags relevant to the forum)
+* Each system that uses the tagging package must have tag editing permissions specific to the system.
```

Comment:
Permissions can be on a tag basis too, in other words a user can have access to only a subset of tags.